### ISSUE TITLE
Platform: Establish a clear heuristic for underlined hyperlinks

### ISSUE THREAD

**Issue by jasmussen** [Automattic/wp-calypso#99254]
There are a lot of instances of basic hypertext inside paragraphs, that's missing underlines:

![Image](https://github.com/user-attachments/assets/a490475c-4f71-4274-8c72-e4e9deb76a15)

![Image](https://github.com/user-attachments/assets/d22ced6e-e42a-4311-88a6-2dea0fe1b608)

![Image](https://github.com/user-attachments/assets/a4e09cef-4aed-4fbd-aab8-5716acb359a3)

![Image](https://github.com/user-attachments/assets/5e22ce42-1e26-4e7e-81ac-ffde5faf4a53)

In these cases, mix of links and text makes it unclear what's clickable, especially for someone with reduced vision. The rule is coming from here:

![Image](https://github.com/user-attachments/assets/b1f17dea-ccf7-4c43-9ed5-6853c55ec8fe)

A simple fix would be to either remove that base rule, and instead apply `text-decoration: none;` situationally. For example in the following example, each tab, title, card, and button component should come with this style:

![Image](https://github.com/user-attachments/assets/1d6506fa-0a86-48d2-8a66-304dfbd17752)

The inverse is also possible keep the removal in place, and add something similar right next to it, such as just underlining links inside paragraphs:

```
p a {
	text-decoration: underline;
}
```

Arguably however, _adding CSS where needed_, is better code quality than adding resets and overrides. In addition to this, there's a systems discussion (CC: @Automattic/dotcom-design) about establishing some clear guidelines for when to underline links, and when not to. GitHub's interface is fine example starting point:

![Image](https://github.com/user-attachments/assets/2b225e0d-13eb-48b6-b701-d791a226a792)

Success is CSS inheritance that's obvious and simple. Failure is a propogation of `text-decoration: underline;` CSS across the codebase.

**Comment by jasmussen** [comment]
An argument for removing the `a { text-decoration: none; }` rule is that it's the inverse of expectations. Which means when components like this Jetpack module is inserted in a context where it inherits this rule, the underline for the link is missing:

![Image](https://github.com/user-attachments/assets/bb02d2a3-0bb2-4aa7-8f97-3f2c2802a837)

**Comment by matt-west** [comment]
Thanks for calling this out @jasmussen.

My two cents on this:

* Links within body copy should be underlined to maintain accessibility.
* Links that are part of navigational components (sidebar menus, breadcrumbs, cards, etc.) should not be underlined.

I agree with you that removing the global override so that _all_ links are underlined until we explicitly override that behaviour is a cleaner approach. It prioritises accessibility in the default behaviour.

I’m not sure how long that global override has been in place, but I assume that removing it will be a pretty big job. We need to carefully test every page to identify areas where we need to apply local overrides. That might also be a good opportunity to transition some legacy code to use components from the design system at the same time though.

**Comment by jasmussen** [comment]
That heuristic sums it up for me 👍 👍

> I’m not sure how long that global override has been in place, but I assume that removing it will be a pretty big job. We need to carefully test every page to identify areas where we need to apply local overrides. That might also be a good opportunity to transition some legacy code to use components from the design system at the same time though.

Oldest I can find is #37064, but that seems to have just moved the "reset rules" to there, from another place. It seems likely that the rule has been in place since the initial commit.

I have a preference for fixing this right, and removing the reset rule. I'm happy to try a PR to that effect, but it'll have to be next week after my support rotation. Though, it's also on my mind to first help land your color studio PR, so we don't open too many cans at once!

**Comment by fditrapani** [comment]
> That heuristic sums it up for me

Plus one on this! Our current approach of using colour in the body section makes it hard to tell links apart from regular content. There isn't enough contrast between the colours.

**Comment by nuriapenya** [comment]
> Links within body copy should be underlined to maintain accessibility.
Links that are part of navigational components (sidebar menus, breadcrumbs, cards, etc.) should not be underlined.

This is the modus operandi that we've been following on the Onboarding UI revision (p9Jlb4-fMP-p2), so totally +1 on following this. 

![Image](https://github.com/user-attachments/assets/11f42d8d-0cca-4637-90f8-0efdd86ae129)

**Comment by tellthemachines** [comment]
I got a little draft PR started to look into how doable it is to remove the base rule: https://github.com/Automattic/wp-calypso/pull/101358

**Comment by tellthemachines** [comment]
Given [#101358](https://github.com/Automattic/wp-calypso/pull/101358) is now merged, can we close this issue?

**Comment by jasmussen** [comment]
Tentatively yes, and thank you. @Automattic/dotcom-design do we need a guideline/best practice doc somewhere that captures this? Or is a proper default (as now handled) sufficient?

**Comment by matt-west** [comment]
> do we need a guideline/best practice doc somewhere that captures this?

I think that would be useful, especially for new designers that are onboarding into the team.

I’m not sure the best place to put it though. Maybe a page in the Dotcom Design P2?

**Comment by jasmussen** [comment]
Pending a better place to archive these docs, I've added a new such section in Figma: dhW7zYBamXJIeJIE3wy5f4-fi-882_21203

With that captured, I'm closing this one out. Thanks for all the work!

**Comment by jasmussen** [comment]
Noting as followup, that we're tracking followups in [Linear](https://linear.app/a8c/project/platform-adhere-to-a-systematic-approach-to-underlined-hyperlinks-ace04dcd00c1/overview) now.
